src/main/java/org/codehaus/plexus/util/xml/Xpp3Dom.java

```diff
@@ -86,6 +88,12 @@ public Xpp3Dom( String name )
         childMap = new HashMap<String, Xpp3Dom>();
     }
 
+    public Xpp3Dom( String name, Object inputLocation )
```

Comment:
Thinking out loud: Can we add a generic to Xpp3Dom, like `Xpp3Dom<T>`, where `T` reflects the inputLocation?

Reply:
I suppose it is feasible: IMHO, gives too much visibility to input location tracking, which is really something very special, but feasible